Include/httpClient/httpClient.h

```diff
@@ -558,6 +547,159 @@ HCHttpCallResponseGetHeaderAtIndex(
+    ) HC_NOEXCEPT;
+
+/// <summary>
+/// Set a header for the WebSocket
+/// This must be called prior to calling HCWebsocketConnect.
+/// </summary>
+/// <param name="websocket">The handle of the WebSocket</param>
+/// <param name="headerName">Header name for the WebSocket</param>
+/// <param name="headerValue">Header value for the WebSocket</param>
+/// <returns>Result code for this API operation.  Possible values are HC_OK, HC_E_INVALIDARG, HC_E_OUTOFMEMORY, or HC_E_FAIL.</returns>
+HC_API HC_RESULT HC_CALLING_CONV
+HCWebSocketSetHeader(
+    _In_ HC_WEBSOCKET_HANDLE websocket,
+    _In_z_ PCSTR headerName,
+    _In_z_ PCSTR headerValue
+    ) HC_NOEXCEPT;
+
+
+/// <summary>
+/// A callback invoked every time a WebSocket receives an incoming message
+/// </summary>
+/// <param name="websocket">Handle to the WebSocket that this message was sent to</param>
+/// <param name="incomingBodyString">Body of the incoming message as a string value, only if the message type is UTF-8.</param>
+typedef void
+(HC_CALLING_CONV* HC_WEBSOCKET_MESSAGE_FUNC)(
+    _In_ HC_WEBSOCKET_HANDLE websocket,
+    _In_z_ PCSTR incomingBodyString
+    );
+
+/// <summary>
+/// A callback invoked when a WebSocket is closed
+/// </summary>
+/// <param name="websocket">Handle to the WebSocket</param>
+/// <param name="closeStatus">The status of why the WebSocket was closed</param>
+typedef void
+(HC_CALLING_CONV* HC_WEBSOCKET_CLOSE_EVENT_FUNC)(
+    _In_ HC_WEBSOCKET_HANDLE websocket,
+    _In_ HC_WEBSOCKET_CLOSE_STATUS closeStatus
+    );
+
+/// <summary>
+/// Sets the WebSocket functions to allow callers to respond to incoming messages and WebSocket close events.
+/// </summary>
+/// <param name="messageFunc">A pointer to the message handling callback to use, or a null pointer to remove.</param>
+/// <param name="closeFunc">A pointer to the close callback to use, or a null pointer to remove.</param>
+HC_API HC_RESULT HC_CALLING_CONV
+HCWebSocketSetFunctions(
+    _In_opt_ HC_WEBSOCKET_MESSAGE_FUNC messageFunc,
+    _In_opt_ HC_WEBSOCKET_CLOSE_EVENT_FUNC closeFunc
+    ) HC_NOEXCEPT;
+
+/// <summary>
+/// Struct for WebSocket connection args
+/// </summary>
+typedef struct HC_WEBSOCKET_CONNECT_INIT_ARGS_STRUCT
+{
+    // TODO: add thread affinity options for platforms need it.  UWP and XDK do not 
+} HC_WEBSOCKET_CONNECT_INIT_ARGS;
+
+/// <summary>
+/// Connects to the WebSocket.  This API returns immediately and will spin up a thread under the covers.
+/// On UWP and XDK, the thread is owned and controlled by Windows::Networking::Sockets::MessageWebSocket
+/// </summary>
+/// <param name="uri">The URI to connect to</param>
+/// <param name="websocket">The handle of the WebSocket</param>
+/// <param name="args">Struct for describing the WebSocket connection args</param>
+/// <returns>Result code for this API operation.  Possible values are HC_OK, HC_E_INVALIDARG, HC_E_OUTOFMEMORY, or HC_E_FAIL.</returns>
+HC_API HC_RESULT HC_CALLING_CONV
+HCWebSocketConnect(
```

Comment:
>HCWebSocketConnect [](start = 0, length = 18)

Connect and send should be async, close is technically async, but we can fire and forget.

Reply:
Updated